controllers/machinehealthcheck_controller.go

```diff
@@ -425,17 +446,13 @@ func (r *MachineHealthCheckReconciler) indexMachineByNodeName(object runtime.Obj
 // isAllowedRemediation checks the value of the MaxUnhealthy field to determine
 // whether remediation should be allowed or not
 func isAllowedRemediation(mhc *clusterv1.MachineHealthCheck) bool {
-	if mhc.Spec.MaxUnhealthy == nil {
-		return true
-	}
-	maxUnhealthy, err := intstr.GetValueFromIntOrPercent(mhc.Spec.MaxUnhealthy, int(mhc.Status.ExpectedMachines), false)
+	maxUnhealthy, err := getMaxUnhealthy(mhc)
 	if err != nil {
 		return false
 	}
 
 	// If unhealthy is above maxUnhealthy, short circuit any further remediation
-	unhealthy := mhc.Status.ExpectedMachines - mhc.Status.CurrentHealthy
-	return int(unhealthy) <= maxUnhealthy
+	return unhealthyMachineCount(mhc) <= maxUnhealthy
```

Comment:
this isn't from this PR but shouldn't remediation only be allowed if unhealthy is strictly less than maxUnhealthy? If unhealthy == maxUnhealthy and further remediation is attempted, wouldn't that risk n + 1 machines to become unhealthy ?

Reply:
This [isn't the cleanest wording](https://github.com/kubernetes-sigs/cluster-api/blob/d4101d79f33001db88e2ca48dc485ed6ccaa8727/api/v1alpha3/machinehealthcheck_types.go#L43-L44) in the English language but

> Any further remediation is only allowed if at most "MaxUnhealthy" machines selected by "selector" are not healthy.

I would say this means "Remediation is allowed if `MaxUnhealthy` machines are unhealthy."

Reply:
I agree the wording isn't the best, but I think Ben hit the nail on the head here, I think the intention of this field is that if `maxUnhealthy: 3` and the current number of unhealthy machines is 3, those three machines can be remediated, however, if 4 were unhealthy, none of the unhealthy machines would be remediated.

This PR just refactors the logic, nothing should have changed from the behaviour here (or at least that was my intention)